src/net/DNS/DNSManager.cs

```diff
@@ -48,6 +54,62 @@ public LookupRequest(string hostname, QType queryType, int timeout, List<IPEndPo
                 DNSServers = dnsServers;
                 CompleteEvent = completeEvent;
             }
+
+            public string id()
+            {
+                return string.Format("{0}{1}", this.QueryType.ToString(), this.Hostname);
+            }
+            #region IDisposable Support
+            private bool disposedValue = false; // To detect redundant calls
+
+            protected virtual void Dispose(bool disposing)

```

Comment:
Why do we need to dispose all of these objects? They seem like managed objects.

Reply:
Because I want to control, when objects are deleted. 
Further more I don't want to raise exceptions on threads waiting for (garbage collected) events, rather than give them a chance to continue the way it was intended by the waiting thread.